Wip

andela-iikikin opened the pull request:
# Get Peer Review for my Room Space Allocator
## What does this PR do?

- **Testsuits to test the following functionalities in room space allocator program**

- [ ] Creates rooms in Dojo.  The user should be able to create as many rooms as possible by specifying multiple room names after the create_room command.

- [ ] Adds a person to the system and allocates the person to a random room. wants_accommodation is an optional argument which can be either Y or N. The default value if it is not provided is N.

- [ ] Prints  the names of all the people in a room on the screen.

- [ ] Prints a list of allocations onto the screen or writes them into a file if an optional file name is given.

- [ ]  Prints a list of people whose allocations dd not succeed to the screen or writes them into a file if an optional file name is given.
## How should this be manually tested?

- Instructions have been included in  the README file.

I-chat committed:
add create_office testcase

I-chat committed:
add more testcases

I-chat committed:
[Fix bug] bj_living not an office object but a string

I-chat committed:
Add testcase for print_allocations and print_room

I-chat committed:
Implement docopt

I-chat committed:
Edit readme file

I-chat committed:
Add test for print_allocated method

I-chat committed:
Fix test_create_room bug

I-chat committed:
Add create_room functionality

I-chat committed:
Add person, staff, fellow class

I-chat committed:
Add room, office, living space class

I-chat committed:
Fix double room bug in create_room

jubrilissa commented on README.md:
```diff
@@ -1,5 +1,5 @@
+Room_space_allocator is a python program that allocates room based on different designations. 
 
 ## Introduction
 * **`Room_space_allocator`** is a Python command line program that allocates diffrent rooms to different persons based on their designation.
```
Update you spelling of *diffrent*. Or you can rephrase the sentence above as. **`Room_space_allocator`** is a Python command line program that allocates rooms to persons based on their designation.

andela-iikikin replied:
Thanks, for the feedback. I will definitely be make these changes.

jubrilissa commented on app/dojo.py:
```diff
@@ -0,0 +1,34 @@
+    all_rooms = {}
+    dojo_persons = {}
+    all_office = []
+    all_ls = []
```
You might want to use a more descriptive name here.

andela-iikikin replied:
OK, will longer names suffice?

jubrilissa commented on tests/room_tests.py:
```diff
@@ -1,9 +1,147 @@
+        my_dojo = Dojo()
+        initial_room_count = len(my_dojo.all_rooms)
+        self.assertEqual(initial_room_count, 0)
+        meeting_office = my_dojo.create_room('office', ['meeting'])
```
Where are you using the meeting_office?

jubrilissa commented on tests/room_tests.py:
```diff
@@ -1,9 +1,147 @@
+        my_dojo = Dojo()
+        initial_room_count = len(my_dojo.all_rooms)
+        self.assertEqual(initial_room_count, 0)
+        blue_ls = my_dojo.create_room('living', ['blue'])
```
Can't seem to find where blue_ls is used.

I-chat committed:
Update file with list of functionalities

I-chat committed:
Change variable and add files to gitignore

Change variable names to a more descriptive names and add .pyc files globally in the gitignore file.

I-chat committed:
Add person ID functionality

I-chat committed:
Remove unused variable names

innocentamadi commented on app/dojo.py:
```diff
@@ -0,0 +1,34 @@
+        for room in room_name:
+            if room in Dojo.all_rooms:
+                output = output + ('Room already exist.\n')
+            elif not room.isalnum():
```
Can the function `isalnum()` be more descriptive?

andela-iikikin replied:
It's a Python inbuilt function. Maybe I will be adding a comment right on top of it. Thank you.

innocentamadi commented on app/dojo.py:
```diff
@@ -0,0 +1,34 @@
+            elif room_type != 'living' and room_type != 'office':
+                output = output + ('Invalid command.\nFirst argument must be either'
+                                   'an office or living.\n')
+        return output
```
Seems like `create_room()` is doing a lot of things. It is concerned with creating all kinds of rooms and handles sanity checks (naming convention verifications).

Heard of the single responsibility principle? A function should do just one thing. Keep it simple, stupid.

This looks like a good opportunity to create a class. The `RoomCreator` class could have a create room public function that delegates to other functions depending on the type of room being created.

andela-iikikin replied:
Yeah. Thanks for that. will break the method in two.

innocentamadi commented on app/person.py:
```diff
@@ -0,0 +1,23 @@
+
+    def __init__(self, first_name, last_name):
+        super(Fellow, self).__init__(first_name, last_name)
+        self.person_type = 'fellow'
```
Excellent class use, documentation and explicit variable naming 👍

andela-iikikin replied:
Thank you.

I-chat committed:
Correct one response for creating more than one room

I-chat committed:
Split add_fellow and add_staff test

I-chat committed:
Edit print_allocations test with different output

I-chat committed:
Add get_available_room for add_person functionality

I-chat committed:
Add setUp and tearDown functions

I-chat committed:
Add arg check for add_person functionality

I-chat committed:
Add add_fellow and add_staff functionality

I-chat committed:
Add test for load_people

I-chat committed:
Correct markdown error

I-chat committed:
Add print_room functionality

I-chat committed:
Add functionalities for print_allocations

I-chat committed:
Add folder for storing app files

I-chat committed:
Delete redundant files

I-chat committed:
Correct all dictionaries mapping object_attr to object

I-chat committed:
Add id_generator function

I-chat committed:
Remove id_generator function

I-chat committed:
Change self to cls in class property

I-chat committed:
Correct instantiation Dojo

I-chat committed:
Add printing unallocated functionality

I-chat committed:
Add functionality for reallocating people

I-chat committed:
Add load people functionalities

I-chat committed:
Add test for save state functionlity

I-chat committed:
Add functionality for load people and save state

I-chat committed:
Add sqlalchemy schema for save state functionalities

I-chat committed:
Fix PEP8 style errors

jubrilissa commented on database/database.py:
```diff
@@ -0,0 +1,59 @@
+    __tablename__ = 'person'
+
+    person_id = Column(String, primary_key=True)
+    firstname = Column(String)
```
for consistency with other parts of your app (e.g your Person and Dojo class). you might want to use snake case for first_name.last_name and full_name

jubrilissa commented on app/dojo.py:
```diff
@@ -0,0 +1,395 @@
+            return(fellow_id)
+
+    @classmethod
+    def add_fellow(cls, first_name, last_name,
```
You might also want to break down this method. Looks like it is doing too much.

jubrilissa reviewed (commented):
I like the way you are generally adhering to pep8 style guide in your code. 👍

I-chat committed:
Add test for save_state and load_state

I-chat committed:
Add functionalities for save_state and load_state

I-chat committed:
Refactor database schema to PickleType

I-chat committed:
Move file to base directory

Move the main.py file from the app folder to the root folder.

I-chat committed:
Add docopt support for load_sate

I-chat committed:
Change test file name

I-chat committed:
Fix reallocate and allocations bug

I-chat committed:
Add directories and more support for reallocate_person

Add database and data directories and also add support for reallocating people on the waiting list in the reallocate_person function.

I-chat committed:
Add support for remove_person and print_all_persons

I-chat committed:
Add tests for remove_person and print_all_persons

I-chat committed:
Add docopt support for remove_person and print_all_persons

I-chat committed:
Add functionalities for remove_person and print_all_persons

I-chat committed:
Add new functionalities description

I-chat committed:
Change all room name to small case

I-chat committed:
Refactor remove_person into two functions

I-chat committed:
Add sqlalchemy to list of dependencies

I-chat committed:
Refactor add_fellow function

I-chat committed:
Delete reduntant file

I-chat committed:
Add tests to increase test coverage

I-chat committed:
Add files to support coverall badges

I-chat committed:
Add travis and coveralls badges

I-chat committed:
Add coveralls badge

I-chat committed:
Refactor methods to follow DRY rule.

I-chat committed:
Correct database name error.

I-chat committed:
Remove redundant attributes and refactor code

njirap commented on .gitignore:
```diff
@@ -34,9 +34,9 @@ __pycache__
 ehthumbs.db
-Thumbs.db
\ No newline at end of file
+Thumbs.db
```
what are these?

njirap commented on app/person.py:
```diff
@@ -0,0 +1,32 @@
+        """Initialize Person object with set variable names."""
+        self.first_name = first_name
+        self.last_name = last_name
+        self.assigned_room = {}
```
`self.assigned_room = {'my_office':'', 'my_livingspace':''}`

njirap commented on data/.gitignore:
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore
```
A README of what this folder is about makes more sense..

njirap commented on database/.gitignore:
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore
```
Same here as comment above.

njirap commented on main.py:
```diff
@@ -0,0 +1,168 @@
+        """Usage: create_room <room_type> <room_name>..."""
+        room_name = arg["<room_name>"]
+        room_type = arg["<room_type>"].lower()
+        if room_type == 'office' or room_type == 'living':
```
`if room_type in ['office', 'living']`

njirap commented on main.py:
```diff
@@ -0,0 +1,168 @@
+        first_name = arg["<person_first_name>"].lower()
+        last_name = arg["<person_last_name>"].lower()
+        person_type = arg["<designation>"].lower()
+        wants_accomodation = arg["<--wants_accomodation>"]
```
`wants_accomodation = arg["<--wants_accomodation>"] or 'n'`

njirap commented on main.py:
```diff
@@ -0,0 +1,168 @@
+        wants_accomodation = arg["<--wants_accomodation>"]
+        if wants_accomodation is None:
+            wants_accomodation = 'n'
+        if (wants_accomodation == 'y' or wants_accomodation == 'n'):
```
`if wants_accomodation.lower() in ['y','n']`
